torch/onnx/_internal/fx/decomposition_table.py

```diff
@@ -0,0 +1,100 @@
+
+import torch
+import torch._decomp
+import torch._ops
+import torch.fx
+from torch.onnx._internal import _beartype
+
+from torch.onnx._internal.fx import registration
+
+
+@_beartype.beartype
+def _create_onnx_supports_op_overload_table(
+    registry: registration.OnnxRegistry,
+) -> Set[Union[torch._ops.OpOverload, Callable]]:
+    """
+    Creates a set of OpOverload and Callable objects that represent ONNX-supported PyTorch operations.
+
+    Args:
+        registry (OnnxRegistry): The ONNX registry for PyTorch.
+
+    Returns:
+        A collection of OpOverload and Callable objects representing ONNX-supported PyTorch operations.
+    """
+    table: Set[Union[torch._ops.OpOverload, Callable]] = set()
+
+    # Some ops in `torch.ops.aten` are not discoverable through `dir(torch.ops.aten)`,
+    # but retrievable via explicit lookup.
+    # https://github.com/pytorch/pytorch/issues/99681
+    # This is a workaround to make sure we register ONNX symbolic functions for these.
+    onnx_supported_aten_lookup_table = [
+        k.split("::")[1].split(".")[0]
+        for k in registry.all_functions()
+        if k.startswith("aten::")
+    ]
+
+    for op_namespace in (torch.ops.aten, torch.ops.prims):
+        attr_names = dir(op_namespace)
+        if op_namespace is torch.ops.aten:
+            attr_names += onnx_supported_aten_lookup_table
+        for attr_name in attr_names:
+            if not hasattr(op_namespace, attr_name):
+                # torchlib owns some attributes that are not aten ops.
+                continue
+            op_overload_packet = getattr(op_namespace, attr_name)
+            if not isinstance(op_overload_packet, torch._ops.OpOverloadPacket):
+                continue
+
+            exporter_look_up_key = op_overload_packet._qualified_op_name
+            if registry.get_function_group(exporter_look_up_key) is None:
+                # This aten op doesn't have ONNX overloads.
+                continue
+
+            for overload_name in op_overload_packet.overloads():
+                op_overload = getattr(op_overload_packet, overload_name)
+                # This line maps torch.ops.aten.add.Tensor, torch.ops.aten.add.Scalar, torch.ops.aten.add.out, etc
+                # to "aten::add". This means the exporter for "aten::add" is used for all overloads of "aten::add".
+                # This is applied to all ops under torch.ops.aten.
+                table.add(op_overload)
+    return table
+
+
+@_beartype.beartype
+def create_onnx_friendly_decomposition_table(
+    registry: registration.OnnxRegistry,
+) -> Dict[torch._ops.OpOverload, Callable]:
+    """
+    This function creates a dictionary of op overloads and their decomposition functions
+    for ops that do not have ONNX symbolic functions. If an op already has an ONNX symbolic function,
+    its decomposition function is excluded from the table. The decomposition table is a subset of PyTorch's
+    built-in aten-to-aten decomposition.
```

Comment:
I incline to not touch the logics here (this was here), as it's not in this PR scope. I know it would be great if we can really make multiple opsets refactor easier, if this PR can do this and do that. But all I really want to focus in this PR is to dispatch the overloads I mentioned in the description.